posts/2024/farmOS-3.1.0/index.md

```diff
@@ -0,0 +1,114 @@
+---
+title: farmOS 3.1.0
+date: 2024-02-07
+author: Michael Stenta
+slug: 2024/farmOS-3.1.0
+categories:
+  - newsletter
+---
+
+# farmOS 3.1.0
+
+Today we are excited to announce the release of farmOS 3.1.0! This release
+includes a number of new features and improvements. Here are some of the
+highlights.
+
+## Product asset type
+
+A new Product asset type has been added, which will allow tracking of "products"
+separate from the other asset types that produce them. Products, like all asset
+types, can be moved between locations, referenced by logs, and have inventory
+levels that are incremented/decremented by activities on or off the farm.
+
+Imagine harvesting a Plant asset and incrementing one or more Product assets
+with the quantity that was harvested! Products can be used to maintain produce
+or meat availability lists. The sky is the limit! It's a simple but powerful
+addition to the [core farmOS data model](https://farmOS.org/model)!
+
+This has been a long time coming, but it is only a first step. This adds the
+data architecture, but work still needs to be done to make it really nice and
+usable. Some work is already underway, and discussions are ongoing in the
+[farmOS forum](https://farmOS.discourse.group). If you have ideas, share them!
+
+## Inventory quick form(s)
+
+A new Inventory quick form is now available, which makes it easier to maintain
+asset inventory levels. Just select the asset you want to adjust and enter the
+quantity (as a "reset", "increment", or "decrement")! There are some more
+advanced options too, which are described in a dedicated post:
+[Inventory Quick Form](inventory-quick-form)
```

Comment:
link doesn't work, on deploy preview it points to `/blog/inventory-quick-form` but the post is at `/blog/2024/inventory-quick-form`

Reply:
Hrm I can't seem to get this to work right. :-(

I tried changing it to `/blog/2024/inventory-quick-form` but that resulted in a link to `/blog/blog/2024/inventory-quick-form`.

I also tried `./inventory-quick-form` (in case the `./` was helpful) but that had the same effect as the original issue (link to `/blog/inventory-quick-form`.

So it seems we may have a bug in the farmOS.org codebase's link generator, but only for blog posts? :thinking:

Bummer. Well for the sake of this PR, I suppose we should just decide how it *should* work in blog posts. I think relative linking makes sense, right? So then we just need to figure out why it isn't being generated correctly in farmOS.org's code.

Alternatively, I could change it to an absolute link to `https://farmOS.org/blog/2024/inventory-quick-form`, but that means the preview wouldn't work, and it feels like it could cause future issues if we ever move things around.

Reply:
I also just noticed (relatedly) that the second link in this 3.0.0 post is broken in the same way on the live site: https://farmos.org/blog/2024/farmOS-3.0.0/

Reply:
https://github.com/farmOS/farmOS.org/issues/90

Reply:
Using post slugs instead of folder names in relative links seems to generate working links:
https://github.com/wotnak/farmOS-community-blog/pull/6/commits/e689094f54a976097111db0e0d010abfae2fc22e 
https://deploy-preview-6--wotnak-farmos-blog-preview.netlify.app/blog/2024/farmOS-3.1.0/

Reply:
Oh good find @wotnak! Thanks! I included your commit in the PR. I'll make note of that in https://github.com/farmOS/farmOS.org/issues/90 too.